core/services/assets/I18nRegistry.php

```diff
@@ -0,0 +1,241 @@
+     *
+     * @param array $handles Array of registered script handles.
+     * @return array
+     */
+    public function queueI18n(array $handles)
+    {
+        if (empty($this->registered_i18n) || empty($this->i18n_map)) {
+            return $handles;
+        }
+        foreach ($handles as $handle) {
+            $this->queueI18nTranslationsForHandle($handle);
+        }
+        if ($this->queued_handle_translations) {
+            foreach ($this->queued_handle_translations as $handle => $translations_for_domain) {
+                $this->registerInlineScript(
+                    $handle,
+                    $translations_for_domain['translations'],
+                    $translations_for_domain['domain']
+                );
+            }
+        }
+        return $handles;
+    }
+
+
+    /**
+     * Registers inline script with translations for given handle and domain.
+     *
+     * @param string $handle       Handle used to register javascript file containing translations.
+     * @param array  $translations Array of string translations.
+     * @param string $domain       Domain for translations.  If left empty then strings are registered with the default
+     *                             domain for the javascript.
+     */
+    private function registerInlineScript($handle, array $translations, $domain = '')
+    {
+        $script = $domain ?
+            'wp.i18n.setLocaleData( ' . json_encode($translations) . ', ' . $domain . ' );' :
+            'wp.i18n.setLocaleData( ' . json_encode($translations) . ' );';
+        wp_add_inline_script($handle, $script, 'before');
+    }
+
+
+    /**
+     * Queues up the translation strings for the given handle.
+     *
+     * @param string $handle The script handle being queued up.
+     */
+    private function queueI18nTranslationsForHandle($handle)
+    {
+        if (isset($this->registered_i18n[$handle])) {
+            $domain = $this->registered_i18n[$handle];
+            $translations = $this->getJedLocaleDataForDomainAndChunk($handle, $domain);
+            if (count($translations) > 1) {
+                $this->queued_handle_translations[$handle] = array(
+                    'domain'       => $domain,
+                    'translations' => $translations,
+                );
+            }
+            unset($this->registered_i18n[$handle]);
+        }
+    }
+
+
+    /**
+     * Sets the internal i18n_map property.
+     * If $chunk_map is empty or not an array, will attempt to load a chunk map from a default named map.
+     *
+     * @param array $i18n_map  If provided, an array of translation strings indexed by script handle names they
+     *                         correspond to.
+     */
+    private function setI18nMap(array $i18n_map)
+    {
+        if (empty($i18n_map)) {
+            $i18n_map = file_exists($this->domain->pluginPath() . 'translation-map.json')
```

Comment:
It's only used here.  I'd rather not implement a constant yet for this because some of this api could change depending on what ends up happening in wp core/gutenberg with the pull request I'm doing.  So much of this could end up getting stripped out once something lands in wp core.